jwk/jwt_strategy.go

```diff
@@ -109,49 +111,60 @@ func (j *RS256JWTStrategy) GetPublicKeyID(ctx context.Context) (string, error) {
 }
 
 func (j *RS256JWTStrategy) refresh(ctx context.Context) error {
-	keys, err := j.r.KeyManager().GetKeySet(ctx, j.rs())
-	if err != nil {
-		return err
-	}
-
-	public, err := FindKeyByPrefix(keys, "public")
-	if err != nil {
-		return err
-	}
-
-	private, err := FindKeyByPrefix(keys, "private")
-	if err != nil {
-		return err
-	}
-
-	if strings.Replace(public.KeyID, "public:", "", 1) != strings.Replace(private.KeyID, "private:", "", 1) {
-		return errors.New("public and private key pair kids do not match")
-	}
-
-	if k, ok := private.Key.(*rsa.PrivateKey); !ok {
-		return errors.New("unable to type assert key to *rsa.PublicKey")
-	} else {
-		j.Lock()
-		j.privateKey = k
-		j.RS256JWTStrategy.PrivateKey = k
-		j.Unlock()
-	}
-
-	if k, ok := public.Key.(*rsa.PublicKey); !ok {
-		return errors.New("unable to type assert key to *rsa.PublicKey")
+	if j.c.HsmEnabled() {
+		if keyPair, err := j.r.HardwareSecurityModule().FindKeyPair(nil, []byte(j.rs())); err != nil {
```

Comment:
Same here, would it be possible to abstract this into `KeyManager`? That way we can avoid `j.c.HsmEnabled` checks wherever we need cryptographic keys